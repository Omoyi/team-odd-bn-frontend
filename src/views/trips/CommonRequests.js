import React, { Component } from 'react';
import { Link } from 'react-router-dom';


class RequestView extends Component {
  render() {
    const {
      data,
      navs,
      tableHeads,
      entities,
      paginationEnd,
      paginationStart,
      paginatedRequest,
      paginationAction,
    } = this.props;

    return (
      <>

        {navs && (
          <div className="tableNav">
            {navs.map((head) => (
              <div className={head.className} key={head.key}>
                {head.attribute}
              </div>
            ))}
          </div>
        )}
        {tableHeads && (
          <div className="tableHead">
            {tableHeads.map((head) => (
              <div key={head.key}>{head.attribute}</div>
            ))}
          </div>
        )}

        {entities
          && entities.map((entity, index) => (
            <div key={index} className="entity">
              {entity.map((i) => (
                <div key={i.key} className={i.className}>
                  {i.attribute}
                </div>
              ))}
            </div>
          ))}
        {entities.length == 0 && (
          <div className="itemsNotFound">
            No trips found click &nbsp;
            <Link to="/trips/oneWay">here</Link>
            &nbsp; to request a new one
          </div>
        )}
        <div className="corasselButtons">
          <div>
            <button
              id="back"
              type="button"
              onClick={() => paginationAction({
                data,
                currentPage: data.slice(paginationStart, paginationEnd),
                paginationStart: paginationStart - 5,
                paginationEnd: paginationEnd - 5,
              })}
            />
          </div>
          <div>
            <button
              id="front"
              type="button"
              onClick={() => paginationAction({
                data,
                currentPage: data.slice(paginationStart, paginationEnd),
                paginationStart: paginationStart + 5,
                paginationEnd: paginationEnd + 5,
              })}
            />
          </div>

        </div>
      </>
    );
  }
}

export default RequestView;
