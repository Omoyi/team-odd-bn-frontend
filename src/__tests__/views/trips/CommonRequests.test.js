import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import CommonRequests from '../../../views/trips/CommonRequests';
import {
  entities,
  navs,
  tableHeads,
  init,
} from '../../../__mocks__/trips/requests';


const mockStore = configureStore([]);
let wrapper;
let store;

describe('Test common Requests', () => {
  it('Test Common Requests functionality', () => {
    store = mockStore(init);
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <CommonRequests data={[[{ a: 'b' }], [{ b: 'a' }]]} entities={entities} navs={navs} tableHeads={tableHeads} />
        </Router>
      </Provider>,
    );
    const page1 = wrapper.find('#page1');
    // const page2 = wrapper.find('#page2');
    const page3 = wrapper.find('#page3');
    page1.simulate('click');
    // page2.simulate('click');
    page3.simulate('click');
    expect(jest.fn().mock.calls).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
