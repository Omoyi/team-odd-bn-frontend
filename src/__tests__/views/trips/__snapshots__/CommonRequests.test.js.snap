// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test common Requests Test Common Requests functionality 1`] = `ReactWrapper {}`;
